### PR TITLE
KTOR-1453 add basic, basic/hash, and form examples

### PR DESCRIPTION
**Pull request by andreyaksenov**
Created separate examples for demonstrating the authentication feature:
- [auth-basic](https://github.com/ktorio/ktor-documentation/tree/auth-examples/codeSnippets/snippets/auth-basic)
- [auth-basic-hash-table](https://github.com/ktorio/ktor-documentation/tree/auth-examples/codeSnippets/snippets/auth-basic-hash-table)
- [auth-form](https://github.com/ktorio/ktor-documentation/tree/auth-examples/codeSnippets/snippets/auth-form)

### CODE REVIEW

File: codeSnippets/snippets/auth-basic-hash-table/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Basic authentication with UserHashedTableAuth
+
+A sample project demonstrating how to use [basic HTTP authentication](https://ktor.io/docs/basic.html) when user names and password hashes are stored in an in-memory table.
```
Comment (Stexxe) [inline codeSnippets/snippets/auth-basic-hash-table/README.md:3]: > when user names and password hashes

What does it mean?

File: codeSnippets/snippets/auth-basic-hash-table/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+## Running
+To run this sample, execute the following command in a repository's root directory:
+```bash
+./gradlew :auth-basic-hash-table:run
```
Comment (Stexxe) [inline codeSnippets/snippets/auth-basic-hash-table/README.md:8]: What if it's a Windows user?

File: codeSnippets/snippets/auth-basic-hash-table/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+```
+
+Then, open [http://localhost:8080/](http://localhost:8080/) and enter one of the credentials specified in  [hashedUserTable](src/main/kotlin/com/example/Application.kt).
```
Comment (Stexxe) [inline codeSnippets/snippets/auth-basic-hash-table/README.md:11]: In a web browser

File: codeSnippets/snippets/auth-basic-hash-table/build.gradle [hunk 4]
```diff
@@ -0,0 +1,18 @@
+plugins {
+    id 'application'
+    id 'org.jetbrains.kotlin.jvm'
+}
+
+mainClassName = "io.ktor.server.netty.EngineMain"
+
+repositories {
+    mavenCentral()
+    maven { url "https://maven.pkg.jetbrains.space/public/p/kotlinx-html/maven" }
```
Comment (Stexxe) [inline codeSnippets/snippets/auth-basic-hash-table/build.gradle:10]: What is this repository for?

File: codeSnippets/snippets/auth-basic/src/main/kotlin/com/example/Application.kt [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import io.ktor.response.*
+import io.ktor.routing.*
+
+fun Application.main() {
+    install(Authentication) {
+        basic("auth-basic") {
+            realm = "Access to the '/' path"
+            validate { credentials ->
+                if (credentials.name == "jetbrains" && credentials.password == "foobar")
+                    UserIdPrincipal(credentials.name)
```
Comment (Stexxe) [inline codeSnippets/snippets/auth-basic/src/main/kotlin/com/example/Application.kt:14]: Better to add curly braces for clarity

**Review by Stexxe — CHANGES_REQUESTED** [review]

File: codeSnippets/snippets/auth-basic-hash-table/README.md (hunk 1 above)
Comment (andreyaksenov) [reply codeSnippets/snippets/auth-basic-hash-table/README.md:3]: Changed to:
> to authenticate users whose credentials are stored in an in-memory table
is it better?

File: codeSnippets/snippets/auth-basic-hash-table/README.md (hunk 2 above)
Comment (andreyaksenov) [reply codeSnippets/snippets/auth-basic-hash-table/README.md:8]: fixed

File: codeSnippets/snippets/auth-basic-hash-table/build.gradle (hunk 4 above)
Comment (andreyaksenov) [reply codeSnippets/snippets/auth-basic-hash-table/build.gradle:10]: copy/paste from another example, removed

File: codeSnippets/snippets/auth-basic/src/main/kotlin/com/example/Application.kt (hunk 5 above)
Comment (andreyaksenov) [reply codeSnippets/snippets/auth-basic/src/main/kotlin/com/example/Application.kt:14]: done

Comment (Stexxe) [reply codeSnippets/snippets/auth-basic/src/main/kotlin/com/example/Application.kt:14]: Please add them for `else` branch too

**Review by Stexxe — APPROVED** [review]

File: codeSnippets/snippets/auth-basic/src/main/kotlin/com/example/Application.kt (hunk 5 above)
Comment (andreyaksenov) [reply codeSnippets/snippets/auth-basic/src/main/kotlin/com/example/Application.kt:14]: done, thanks

**Pull request merged**